browser/js/common/directives/graphs/line-graph.directive.js

```diff
@@ -0,0 +1,115 @@
+app.directive('lineGraph', function(d3Service, $window) {
+    return {
+        restrict: 'EA',
+        scope: {
+            data: "=",
+            cols: "="
+        },
+        link: function(scope, ele, attrs) {
+
+            d3Service.d3().then(function(d3) {
+                let margin = { top: 20, right: 20, bottom: 30, left: 50 },
+                    width = 960 - margin.left - margin.right,
+                    height = 500 - margin.top - margin.bottom,
+                    dataColumns = scope.cols.split(","),
+                    dateTitles = ['year', 'date', 'time'],
+                    dateFormat;
+
+                let filteredData = scope.data.filter(obj => obj[dataColumns[0]] && obj[dataColumns[1]]);
+
+
+
+
+                //check if the data column header may contain date info ??
+                if (dataColumns[0] === 'year') {
+                    console.log("Here");
+                    //if so validate the format of the date
+
+                    //run date checking function
+                    let commonDateFormats = ["%Y", "%Y-%y", "%x", "%xT%X"];
+
+                    dateFormat = commonDateFormats.filter(f => d3.time.format(f).parse(filteredData[0][dataColumns[0]]))[0];
+                    console.log(dateFormat)
+                }
+
+                let svg = d3.select(ele[0])
+                    .append('svg')
+                    .style('width', '100%')
+                    .style('height', "1000px")
+                    .append("g")
+                    .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+                let formatDate = d3.time.format(dateFormat); //d3.time.format("%Y-%y");
+
+                let x = d3.time.scale()
+                    .range([0, width]);
+
+                let y = d3.scale.linear()
+                    .range([height, 0]);
+
+                let xAxis = d3.svg.axis()
+                    .scale(x)
+                    .orient("bottom");
+
+                let yAxis = d3.svg.axis()
+                    .scale(y)
+                    .orient("left");
+
+                let line = d3.svg.line()
+                    .x(function(d) {
+                        return x(formatDate.parse( d[ dataColumns[0] ] ));
```

Comment:
I'm thinking this `dataColumns` array should instead be an object with two keys: `.x` and `.y`. I think that'll make this code easier to read/debug/reason-about.